packages/web-app-search/src/portals/SearchBar.vue

```diff
@@ -152,13 +154,34 @@ export default defineComponent({
       }
     })
 
+    const isShareRoute = () => {
+      return !!shareId.value
+    }
+
     const optionsDrop = computed(() => {
       return unref(optionsDropRef) as InstanceType<GlobalComponents['OcDrop']>
     })
 
     const availableProviders = computed(() => {
       return unref(providerStore)?.availableProviders
     })
+
+    const buildLocationScopeId = () => {
+      const currentFolder = store.getters['Files/currentFolder']
+      const path = currentFolder.path === '/' ? '' : currentFolder.path
+      if (isShareRoute()) {
+        const shareSpaceResource = buildShareSpaceResource({
+          shareId: currentFolder.id,
+          shareName: currentFolder.name,
+          serverUrl: ''
+        })
+        const jailId = shareSpaceResource.id.toString().split('!')[0]
+        return `${jailId}$${jailId}!${shareId.value}${path}`
```

Comment:
Note that we have a the static JAIL_ID in /home/jan/working/web/packages/web-client/src/helpers/space/types.ts, maybe you could use that.